src/vscode_ocaml_platform.ml

```diff
@@ -37,6 +37,37 @@ let activate (extension : ExtensionContext.t) =
   Cm_editor.register extension instance;
   Repl.register extension instance;
   let sandbox_opt = Sandbox.of_settings_or_detect () in
+  ExtensionContext.subscribe
+    extension
+    ~disposable:
+      (Workspace.onDidChangeConfiguration
+         ~listener:(fun _event ->
+           match Extension_instance.language_client instance with
+           | None -> ()
+           | Some client ->
+             let codelens =
+               Option.map
+                 Settings.(get server_codelens_setting)
+                 ~f:(fun enable ->
+                   LanguageClient.OcamllspSetting.create ~enable ())
+             in
+             let extendedHover =
+               Option.map
+                 Settings.(get server_extendedHover_setting)
+                 ~f:(fun enable ->
+                   LanguageClient.OcamllspSetting.create ~enable ())
+             in
+             let options =
+               LanguageClient.OcamllspConfig.create ?codelens ?extendedHover ()
+             in
+             let payload =
+               LanguageClient.DidChangeConfiguration.create ~settings:options ()
+             in
+             LanguageClient.sendNotification
+               client
+               "workspace/didChangeConfiguration"
+               payload)
+         ());
```

Comment:
I'm not sure where this code should live if not here

Reply:
I don't think there's enough here to require it's own module, so I would extract this code into a function in this file.

Also, could the `let sandbox_opt = ...` be moved after this code? It doesn't look like the new additions rely on its value.

Reply:
ok I moved the code to its own function. And indeed it doesn't have to be after `sandbox_opt`, that was a mistake